pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -88,10 +88,41 @@ func (uc *UpgradeCluster) UpgradeCluster(az armhelpers.AKSEngineClient, kubeConf
 	uc.UpgradedMasterVMs = &[]compute.VirtualMachine{}
 	uc.AgentPools = make(map[string]*AgentPoolTopology)
 
-	if err := uc.getClusterNodeStatus(az, uc.ResourceGroup, kubeConfig); err != nil {
+	var kubeClient armhelpers.KubernetesClient
+	if az != nil {
+		timeout := time.Duration(60) * time.Minute
+		k, err := az.GetKubernetesClient("", kubeConfig, interval, timeout)
+		if err != nil {
+			uc.Logger.Warnf("Failed to get a Kubernetes client: %v", err)
+		}
+		kubeClient = k
```

Comment:
should we assume that `k` will have a valid k8s client object in the `err != nil` case?

Reply:
Maybe not, but this code is moved from lower down and I hadn't changed it. I'm still not positive if this should be a fatal error (it wasn't previously when we only used the client as a fallback upgrade strategy).

Reply:
We check for `nil` before using it. I think this is consistent with the previous optional use of the kubeClient, and also with `upgrade --force` as I implemented it here.